### PR TITLE
fix(quickstart): enhance nodejs_quickstart with logging and testing

### PR DESCRIPTION
**Pull request by ybyzek**
Enhances https://github.com/TheNileDev/examples/tree/main/nodejs_quickstart

* Refactors code a bit (import constants, tweaked logs)
* `iteration_id` moved as input parameter instead of inside code
* Adds a new `nodejs_quickstart/src/find-entity.ts` script to validate entity creation

Note: all `*.ts` files moved to `nodejs_quickstart/src/` so the diffs on `index.ts` aren't directly visible in the PR (need to diff [original](https://github.com/TheNileDev/examples/blob/main/nodejs_quickstart/index.ts) and [new](https://github.com/TheNileDev/examples/blob/quickstart-yb/nodejs_quickstart/src/index.ts)).

### CODE REVIEW

File: nodejs_quickstart/README.md [hunk 1]
```diff
@@ -3,14 +3,47 @@
...
+```
+yarn setup 2
+```
+
+## Validate
+
+Run the following command:
+
+```
+yarn test-setup
```
Comment (gwenshap) [inline nodejs_quickstart/README.md:46]: curious, how is this different from `yarn setup`? what does `validate` mean in the context of a quickstart?

File: nodejs_quickstart/README.md [hunk 2]
```diff
@@ -3,14 +3,47 @@
...
+Run the following command:
+
+```
+yarn setup
+```
+
+To run it repeatedly with new entries, pass in a unique parameter that will be the suffix:
+
+```
+yarn setup 2
```
Comment (gwenshap) [inline nodejs_quickstart/README.md:38]: Nice!!!

File: nodejs_quickstart/README.md [hunk 3]
```diff
@@ -3,14 +3,47 @@
...
+
+## Validate
+
+Run the following command:
+
+```
+yarn test-setup
+```
+
+Log into the [Nile Admin Dashboard](https://nad.thenile.dev/) (default username/password: dev-mary@dw.demo/password) to see the control plane and data plane instances.
```
Comment (gwenshap) [inline nodejs_quickstart/README.md:49]: Good touch

File: nodejs_quickstart/src/find-entity.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+import {iteration_id, NILE_URL, NILE_WORKSPACE} from './constants.ts';
+import {NILE_DEVELOPER_EMAIL, NILE_DEVELOPER_PASSWORD} from './constants.ts';
+import {NILE_TENANT_NAME} from './constants.ts';
+
+const nile = Nile({
+  basePath: NILE_URL,
+  workspace: NILE_WORKSPACE,
+});
+
+async function run() {
```
Comment (gwenshap) [inline nodejs_quickstart/src/find-entity.ts:13]: is this file intended as a separate example that can run on its own? or part of the main quickstart (in index.js)? or both? It isn't obvious to me.

**Review by gwenshap — COMMENTED** [review]
Left some comments where I got confused. 

I don't know much about the CI/CD parts, so I can't review that.

File: nodejs_quickstart/README.md (hunk 1 above)
Comment (ybyzek) [reply nodejs_quickstart/README.md:46]: @gwenshap `yarn test-setup` runs https://github.com/TheNileDev/examples/blob/quickstart-yb/nodejs_quickstart/src/find-entity.ts and the intention is to programmatically validate that the entity instance was created, using the API.

So `setup` did a set of actions, `test-setup` validates.  That's the intention anyway.  Happy to rename if something else jives better.

File: nodejs_quickstart/src/find-entity.ts (hunk 4 above)
Comment (ybyzek) [reply nodejs_quickstart/src/find-entity.ts:13]: @gwenshap It's meant to run on its own (`test-setup`), for reasons mentioned in [this comment](https://github.com/TheNileDev/examples/pull/10#discussion_r955520471).

File: nodejs_quickstart/README.md (hunk 1 above)
Comment (gwenshap) [reply nodejs_quickstart/README.md:46]: Got it! Worth explaining this in the readme? This answers both my questions :)

File: nodejs_quickstart/src/find-entity.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+  var maybeTenant = myOrgs.find( org => org.name == NILE_TENANT_NAME)
+  if (maybeTenant) {
+    console.log("Org " + NILE_TENANT_NAME + " exists with id " + maybeTenant.id)
+    tenant_id = maybeTenant.id
+  } else {
+    console.error("Cannot find id for " + NILE_TENANT_NAME)
+  }
+
+  // Find instance with matching name
+  var instances = await nile.entities.listInstances({
```
Comment (gwenshap) [inline nodejs_quickstart/src/find-entity.ts:43]: You are finding an instance, not an entity. So the file name is misleading. It looked like you are validating the wrong step and had to double check.

**Review by gwenshap — APPROVED** [review]
Approved assuming that:
- `find_entity` will be renamed to `find_instance` since this is what you are doing, I believe? 
- Readme will clarify what validation and the relation between the two files.

File: nodejs_quickstart/src/find-entity.ts (hunk 5 above)
Comment (ybyzek) [reply nodejs_quickstart/src/find-entity.ts]: @gwenshap great catch.  Fixed in latest commit
